File: android2_lesson06/src/java/java/com/example/ama/android2_lesson06/ui/mvp/SmsExamplePresenter.java
package com.example.ama.android2_lesson06.ui.mvp;

import android.database.Cursor;

import com.example.ama.android2_lesson06.ui.base.BaseSmsPresenter;
import com.example.ama.android2_lesson06.ui.base.SmsExampleView;

import io.reactivex.android.schedulers.AndroidSchedulers;

public class SmsExamplePresenter<T extends SmsExampleView> extends BaseSmsPresenter<T> {

    @Override
    public void exportMessages(Cursor cursor) {
        disposable.add(manager.startExport(cursor)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(s -> {
                    if (view != null) view.showMessage(s);
                }));
    }

    @Override
    public void importMessages() {
        disposable.add(manager.startImport()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(s -> {
                    if (view != null) view.showMessage(s);
                }));
    }
}

File: android2_lesson06/src/kotlin/java/com/example/ama/android2_lesson06/ui/mvp/SmsExamplePresenter.kt
package com.example.ama.android2_lesson06.ui.mvp

import android.database.Cursor
import com.example.ama.android2_lesson06.ui.base.BaseSmsPresenter
import com.example.ama.android2_lesson06.ui.base.SmsExpampleView
import io.reactivex.android.schedulers.AndroidSchedulers

class SmsExamplePresenter<T : SmsExpampleView> : BaseSmsPresenter<T>() {

    override fun exportMessages(cursor: Cursor?) {
        disposable?.add(manager.startExport(cursor)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { message -> view?.showMessage(message) })
    }

    override fun importMessages() {
        disposable?.add(manager.startImport()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { message -> view?.showMessage(message) })
    }
}